### ISSUE TITLE
Vaporization as heap memory management technique

### ISSUE THREAD

**Issue by dumblob** [mobarski/morty#4]
It's probably too early to look into this, but I'd like to point out an interesting technique which is both simple to implement and has basically no user-level cognitive load associated (unlike ownership stuff like in Rust, unlike manual memory management with `realloc()` `free()`, unlike garbage collection with manually managed cycles due to weak pointers, etc.).

https://slightknack.dev/blog/vaporization/

The downside is it's not suitable for every language - especially because it needs some compile-time guarantees to work correctly and these might not be offered by Morty - IDK :wink:.

**Comment by mobarski** [comment]
That was an interesting read but as You said it might not be for Morty.
For memory allocation I'm thinking about Zone Allocator or Buddy Block Allocator:
https://slembcke.github.io/2020/10/12/CustomAllocators.html

**Comment by dumblob** [comment]
Yep, custom allocators are handy. But actually allocators have nothing to do with Vaporization. Vaporization is solely about the allocated memory **lifetime management** (without any notion about "how to actually allocate & maintain allocated memory") but allocators in contrast have no notion of lifetime and only care about "how to allocate & maintain allocated memory" :wink:.

In other words, allocators can be used with Vaporization simultaneously (if Morty offers enough of semantics & guarantees for Vaporization to work).

Actually heap memory usage is kind of a solved issue nowadays when it comes to allocators. But the lifetime of heap memory allocations is not a solved problem - therefore when I read about Vaporization I thought it might be something interesting for Morty (maybe even up to the point Morty could adjust its semantics constraints to allow for Vaporization - I don't know... it needs investigation).
